smart-contracts/contracts/cl-vault/src/math/liquidity.rs

```diff
@@ -41,16 +41,71 @@ pub fn _liquidity0(
     let intermediate_2 = Decimal256::from_atomics(intermediate, 36).unwrap();
 
     // since we start with Decimal and multiply with big_factor, we expect to be able to convert back here
-    Ok(Decimal::new(intermediate_2.atomics().try_into()?))
+    Ok(Decimal256::new(intermediate_2.atomics()))
+}
+
+/// asset0 calculates the
```

Comment:
There is a lot of common code in the methods of asset0, asset1, and _liquidity0/1 that can be refactored to reuse common parts by using simpler helpers.

Reply:
Agreed, but this way it's a 1:1 copy of the CL formulas from the [Osmosis docs](https://github.com/osmosis-labs/osmosis/tree/main/x/concentrated-liquidity), refactoring would require a deeper understanding of the actual formulas when checking whether we implement the osmosis math